frontend/src/libs/components/header/header.tsx

```diff
@@ -1,30 +1,56 @@
-import { Avatar, Link } from '~/libs/components/components.js';
+import { Avatar, Link, Popover } from '~/libs/components/components.js';
 import { AppRoute } from '~/libs/enums/enums.js';
 import { getFullName } from '~/libs/helpers/helpers.js';
+import { useCallback, useModal } from '~/libs/hooks/hooks.js';
 import { type UserAuthResponseDto } from '~/packages/users/users.js';
 
+import { DropDown } from './dropdown/dropdown.js';
 import styles from './styles.module.scss';
 
 type Properties = {
   user: UserAuthResponseDto | null;
 };
 
-const Header: React.FC<Properties> = ({ user }) => (
-  <>
-    {user && (
-      <header className={styles.header}>
-        <Link to={AppRoute.ROOT} className={styles.logo}>
-          WRITORIUM
-        </Link>
-        <Link to={AppRoute.PROFILE}>
-          <Avatar
-            username={getFullName(user.firstName, user.lastName)}
-            avatarUrl={null}
+const Header: React.FC<Properties> = ({ user }) => {
+  const { handleToggleModalOpen, isOpen } = useModal();
+
+  const handleClickOnAvatar = useCallback((): void => {
+    if (!isOpen) {
+      handleToggleModalOpen();
+    }
+  }, [handleToggleModalOpen, isOpen]);
+
+  return (
+    <>
+      {user && (
+        <div className={styles.outerheader}>
+          <header className={styles.header}>
+            <Link to={AppRoute.ROOT} className={styles.logo}>
+              WRITORIUM
+            </Link>
+
+            <button
+              className={styles.avatarButton}
+              onClick={handleClickOnAvatar}
+            >
+              <Avatar
+                username={getFullName(user.firstName, user.lastName)}
+                avatarUrl={null}
+              />
+            </button>
+          </header>
+
+          <Popover
+            trigger={{ handleToggleModalOpen, isOpen }}
+            content={<DropDown trigger={{ handleToggleModalOpen, isOpen }} />}
+            className={`${styles.dropdown} ${styles.dropdownModal} ${
```

Comment:
Please use `getValidClassnames` helper here

Reply:
Are you sure this is necessary? This className props gets to the Modal component, where getValidClassnames helper is already used. So classes are already correctly recognized.

Reply:
Yep, use this helper for all classes composition

Reply:
Thanks, got it